blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/LocatorTest.java

```diff
@@ -0,0 +1,435 @@
+        // when
+        Locator locator = new Locator();
+
+        // then
+        assertNotNull(locator);
+        assertNull(locator.getMetricName());
+    }
+
+    @Test
+    public void setStringRepSetsTenant() {
+
+        // given
+        Locator locator = new Locator();
+
+        // when
+        locator.setStringRep(fullyQualifiedName);
+
+        // then
+        assertEquals(tenant, locator.getTenantId());
+    }
+
+    @Test
+    public void setStringRepSetsMetricName() {
+
+        // given
+        Locator locator = new Locator();
+
+        // when
+        locator.setStringRep(fullyQualifiedName);
+
+        // then
+        assertEquals(metricName, locator.getMetricName());
+    }
+
+    @Test
+    public void setStringRepSetsStringRepresentation() {
+
+        // given
+        Locator locator = new Locator();
+
+        // when
+        locator.setStringRep(fullyQualifiedName);
+
+        // then
+        assertEquals(fullyQualifiedName, locator.toString());
+    }
+
+    @Test
+    public void singleParamFactorySetsTenant() {
+
+        // when
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // then
+        assertEquals(tenant, locator.getTenantId());
+    }
+
+    @Test
+    public void singleParamFactorySetsMetricName() {
+
+        // when
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // then
+        assertEquals(metricName, locator.getMetricName());
+    }
+
+    @Test
+    public void singleParamFactorySetsStringRepresentation() {
+
+        // when
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // then
+        assertEquals(fullyQualifiedName, locator.toString());
+    }
+
+    @Test
+    public void multiParamFactorySetsStringRepresentation() {
+
+        // given
+        String expected = tenant + ".a.b.c";
+
+        // when
+        Locator locator = Locator.createLocatorFromPathComponents(tenant, "a", "b", "c");
+
+        // then
+        assertEquals(expected, locator.toString());
+    }
+
+    @Test
+    public void multiParamFactorySetsTenantFromFirstParam() {
+
+        // when
+        Locator locator = Locator.createLocatorFromPathComponents(tenant, "a", "b", "c");
+
+        // then
+        assertEquals(tenant, locator.getTenantId());
+    }
+
+    @Test
+    public void multiParamFactorySetsMetricNameFromPathComponents() {
+
+        // given
+        String expected = "a.b.c";
+
+        // when
+        Locator locator = Locator.createLocatorFromPathComponents(tenant, "a", "b", "c");
+
+        // then
+        assertEquals(expected, locator.getMetricName());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void isValidDBKeyThrowsExceptionOnNullDbKey() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // when
+        locator.isValidDBKey(null, "a");
```

Comment:
@ChandraAddala That's not really part of the story. I'll add a comment about it and add a new story to change it.